### ISSUE TITLE
`promise.json()` doesn't throw `ParseError`

### ISSUE THREAD

**Issue by szmarczak** [sindresorhus/got#1069]
```js
const got = require('got');

(async () => {
    try {
        // This works as expected
        // await got('https://example.com', {responseType: 'json'}).json();

        // This is incorrect
        await got('https://example.com').json();
    } catch (error) {
        console.log(error.name);
        // Actual: SyntaxError
        // Expected: Got.ParseError
    }
})();
```

**Comment by szmarczak** [comment]
Added some reproducible code if someone wants to pick this up.

**Comment by szmarczak** [comment]
Fixed in 4fd10068013f0e7c8047c254cf40b3c81cf35b16
